components/styled/table/src/index.tsx

```diff
@@ -1,20 +1,24 @@
-import { FC } from 'react'
+import GenericTable, { TableData } from './atoms/table'
 
-import { Questionnaire, QuestionnaireResponse, SummaryTableViewType } from '@ltht-react/types'
-import QuestionnaireTable from './molecules/questionnaire-table'
-
-const Table: FC<IProps> = ({ definition, records, orientation = 'VERTICAL' }) => {
-  if (!definition.item || definition.item.length === 0) {
+export default function Table<TColumn, TRow>({ columnData, rowData, mapToTableData }: IProps<TColumn, TRow>) {
+  if (
+    !columnData ||
+    !rowData ||
+    (Array.isArray(columnData) && columnData?.length === 0) ||
+    (Array.isArray(rowData) && rowData?.length === 0)
+  ) {
     return <div>Could not render table. Definition items array was empty.</div>
   }
 
-  return <QuestionnaireTable definitionItems={definition.item} records={records} orientation={orientation} />
-}
+  if (typeof mapToTableData !== 'function') {
```

Comment:
Just `if (!mapToTableData)` is sufficient here since you're declaring the type already in the props 
(Also there's a super minor typo in the div - mapping should have a capital M)